File: test/fast_plot_run_levante.py
#!/usr/bin/env python3
#python3 test/fast_plot_run_levante.py
"""
Python rewrite of test/fast_plot_run_levante.jl
Reads `mountain_wave.h5`, plots a 2D pcolormesh and a PDF of the selected field.
"""
import os
from pathlib import Path
import h5py
import numpy as np
import matplotlib.pyplot as plt
from math import sqrt, pi

# Base font-size (points). Change this to scale all plot text.
FS = 12

INPUT = Path.home() /"spr/ice_mountain_wave.h5"
OUT_DIR = Path.home() / "output"
OUT_DIR.mkdir(parents=True, exist_ok=True)

with h5py.File(INPUT, "r") as data:
    # Read grid and fields (units converted to km)
    x_vals = np.array(data["x"]) * 0.001
    y_vals = np.array(data["y"]) * 0.001
    z = np.array(data["z"]) * 0.001  # expected shape (nx, ny, nz)

    n_y = data["w"].shape[2]
    iy = 0 #n_y // 2
    iys = 0
    print("iy =", iy)
    print(z[0,0,:])

    tidx = 20
    fld = np.array(data["u"][tidx, ...])

    print("size fld ", fld.shape)
    print("fld ", np.nanmax(fld), np.nanmin(fld))

    # --- Plot 2D field (x vs z) at slice iy ---
    fig = plt.figure(figsize=(12, 10))
    ax1 = fig.add_subplot(2, 1, 1)

    # Slice arrays to shapes (nx, nz) for plotting
    F = fld[:, iy, :]

    #pcm = ax1.pcolormesh(X, Z, Z, cmap="Blues")
    #pcm = ax1.pcolormesh(x_vals, z[:,iy,:], F, cmap="Blues")
    pcm = ax1.contourf(np.broadcast_to(x_vals[None,:], F.shape), z[:,iy,:], F, levels=10)
    cbar = fig.colorbar(pcm, ax=ax1)
    #cbar.ax.tick_params(labelsize=FS)
    ax1.set_title("data: Res", fontsize=1.5 * FS)
    ax1.set_xlabel("x (km)", fontsize=1.2 * FS)
    ax1.set_ylabel("z (km)", fontsize=1.2 * FS)
    ax1.tick_params(labelsize=FS)

    # The second subplot for the secondary dataset was disabled in the Julia file.
    # We'll leave the second subplot blank to match the structure.
    ax2 = fig.add_subplot(2, 1, 2)
    ax2.axis("off")

    plt.tight_layout()
    fig.savefig(OUT_DIR / "mountain_wave.png")
    plt.close(fig)

File: test/fast_plot_run_levante.jl
# RUN in PF/pinc/
# julia --project=. examples/visualization/fast_plot.jl
# view with: eog examples/visualization/mountain_wave.png &
#
using HDF5
#using LaTeXStrings
using PyPlot
using Statistics

# Base font-size (points). Change this to scale all plot text.
FS = 12

data = h5open("mountain_wave.h5")
#data = h5open("/home/dolaptch/PF/pinc/exp/pincflow_output.h5")
#data = h5open("/home/dolaptch/PF/runs/tjl05/pincflow_output.h5")
#data = h5open("/home/dolaptch/PF/runs_save/GWI_grusi_WS_25-26/tjl05/pincflow_output.h5")

#data2s = h5open("/home/dolaptch/PF/runs_save/GWI_grusi_WS_25-26/tjl06/pincflow_output.h5")
#data2s = h5open("/home/dolaptch/PF/runs_save/GWI_grusi_WS_25-26/tjl06/pincflow_output.h5")

# Set grid for data1
x = data["x"][:] .* 0.001 
y = data["y"][:] .* 0.001
z = data["z"][:, :, :] .* 0.001
x = [xi for xi in x, iy in 1:size(z)[2], iz in 1:size(z)[3]]
y = [yi for ix in 1:size(z)[1], yi in y, iz in 1:size(z)[3]]

# Set grid for data2
# x2 = data2s["x2"][:] .* 0.001 
# y2 = data2s["y2"][:] .* 0.001 
# z2 = data2s["z2"][:, :, :] .* 0.001
# x2 = [xi for xi in x2, iy in 1:size(z2)[2], iz in 1:size(z2)[3]]
# y2 = [yi for ix in 1:size(z2)[1], yi in y2, iz in 1:size(z2)[3]]

iy = Int(length(data["w"][1, :, 1, 1]) / 2)
iys = 1 
println("iy = ", iy)

tidx = length(data["w"][1, 1, 1, :])
#tidx2s = length(data2s["w"][1, 1, 1, :])


fld = data["iaux1"][:, :, :, tidx]
#fld2s = data2s["sn"][:, :, :, tidx2s]

println("size fld  ", size(fld))
#println("size fld2s", size(fld2s))
println("fld  ", maximum(fld), " ", minimum(fld))
##println("fld2s", maximum(fld2s), " ", minimum(fld2s))

# Plot comparison: data vs datas
figure(figsize=(12, 10))
subplot(211)
contour = pcolormesh(x[:, iy, :], z[:, iy, :], fld[:,iy,:]; cmap="Blues")
cbar = colorbar(contour)
cbar.ax.tick_params(labelsize=FS)
title("data: Res", fontsize=1.5*FS)
xlabel("x (km)", fontsize=1.2*FS); ylabel("z (km)", fontsize=1.2*FS)
ax = gca(); ax.tick_params(labelsize=FS)

## subplot(212)
## contour2s = pcolormesh(x2[:, iys, :], z2[:, iys, :], fld2s[:,iys,:]; cmap="Blues")
## cbar2 = colorbar(contour2s)
## cbar2.ax.tick_params(labelsize=FS)
## title("data2s: Par (saved)", fontsize=1.5*FS)
## xlabel("x (km)", fontsize=1.2*FS); ylabel("z (km)", fontsize=1.2*FS)
## ax = gca(); ax.tick_params(labelsize=FS)

tight_layout()
savefig("~/output/mountain_wave.png")
